### ISSUE TITLE
Improve support for PEP-675 (LiteralString)

### ISSUE THREAD

**Issue by bryanforbes** [microsoft/pyright#3756]
Since the PEP has been accepted and typeshed has been updated to accommodate `LiteralString`, it would be nice to get full support for `LiteralString` as specified in [PEP-675](https://peps.python.org/pep-0675/) in the [specification section](https://peps.python.org/pep-0675/#specification) and [appendix C](https://peps.python.org/pep-0675/#appendix-c-str-methods-that-preserve-literalstring).

**Comment by erictraut** [comment]
Is there something specific that you think is missing?

**Comment by bryanforbes** [comment]
I'm running into issues with the following cases:

```python
from __future__ import annotations

from collections.abc import Iterable
from typing_extensions import LiteralString


def f_string(literal_string: LiteralString) -> LiteralString:
    return f'f-string {literal_string}'


def joiner(parts: Iterable[LiteralString]) -> LiteralString:
    return ', '.join(parts)


def complex_joiner(
    parts: Iterable[tuple[LiteralString, LiteralString]]
) -> LiteralString:
    return ' ' + ' '.join(map(lambda items: f'{items[0]} {items[1]}', parts))
```

**Comment by erictraut** [comment]
These will be fixed in the next release.

**Comment by bryanforbes** [comment]
Thanks, @erictraut. I figured `complex_joiner` might not type check properly even with PEP-675 support, so that's not a big deal.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.265, which I just published. It will be included in a future release of pylance.
